.github/workflows/testCommitExceptMain.yml

```diff
@@ -0,0 +1,8 @@
+name: unit tests
+on:
+  push:
+    branches-ignore: [main]
```

Comment:
I like this setup, where you're able to make sure tests pass again before merging to main in the onPushToMain workflow 👍